velox/exec/HashProbe.h

```diff
@@ -75,9 +75,9 @@ class HashProbe : public Operator {
   // producer.
   void clearIdentityProjectedOutput();
 
-  // Populate output columns with build-side rows that didn't match join
+  // Populate output columns with build-side rows based on the join
```

Comment:
Please elaborate the behavior of "based on" in the comment. It would help to give direct cases to be considered and the expected results of them.